docs/content/concepts/partitions-schedules-sensors/sensors.mdx

```diff
@@ -334,6 +394,34 @@ def test_my_directory_sensor_cursor():
         assert validate_run_config(log_file_job, run_request.run_config)
 ```
 
+#### Testing sensors with resources
+
+For sensors which utilize [resources](/concepts/resources), you can provide the necessary resources when invoking the sensor function.
+
+Below is a test for the `process_new_users_sensor` that we defined [above](#using-resources-in-sensors) which uses the `users_api` resource.
+
+```python file=/concepts/resources/pythonic_resources.py startafter=start_test_resource_on_sensor endbefore=end_test_resource_on_sensor dedent=4
+from dagster import build_sensor_context, validate_run_config
+
+def test_process_new_users_sensor():
+    class MockUsersAPI(ConfigurableResource):
+        def fetch_users(self) -> List[str]:
+            return ["1", "2", "3"]
+
+    # You can build resources into the sensor context
+    context = build_sensor_context(
+        resources={"users_api": MockUsersAPI()},
+    )
+    run_requests = process_new_users_sensor(context)
+    assert len(run_requests) == 3
+
+    # You can also supply resources as kwargs to the sensor function
+    context = build_sensor_context()
+    run_request = process_new_users_sensor(context, users_api=MockUsersAPI())
+    run_requests = process_new_users_sensor(context)
+    assert len(run_requests) == 3
+```
+
```

Comment:
let's only pass directly. I think the context-building method should be considered an "advanced" or "legacy" API.